engine/src/main/java/org/pentaho/di/trans/steps/script/ScriptAddedFunctions.java

```diff
@@ -2412,58 +2413,63 @@ public static Object trunc( ScriptEngine actualContext, Bindings actualObject, O
   @SuppressWarnings( "fallthrough" )
   public static Object truncDate( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
     Object FunctionContext ) {
-    try {
-      // 2 arguments: truncation of dates to a certain precision
-      //
-      if ( ArgList.length == 2 ) {
-        if ( isNull( ArgList[0] ) ) {
-          return null;
-        } else if ( isUndefined( ArgList[0] ) ) {
-          return undefinedValue;
-        }
-
-        // This is the truncation of a date...
-        // The second argument specifies the level: ms, s, min, hour, day, month, year
-        //
-        java.util.Date dArg1 = (java.util.Date) ArgList[0];
-        Calendar cal = Calendar.getInstance();
-        cal.setTime( dArg1 );
-
-        Integer level = (Integer) ArgList[1];
-
-        switch ( level.intValue() ) {
-        // MONTHS
-          case 5:
-            cal.set( Calendar.MONTH, 1 );
-            // DAYS
-          case 4:
-            cal.set( Calendar.DAY_OF_MONTH, 1 );
-            // HOURS
-          case 3:
-            cal.set( Calendar.HOUR_OF_DAY, 0 );
-            // MINUTES
-          case 2:
-            cal.set( Calendar.MINUTE, 0 );
-            // SECONDS
-          case 1:
-            cal.set( Calendar.SECOND, 0 );
-            // MILI-SECONDS
-          case 0:
-            cal.set( Calendar.MILLISECOND, 0 );
-            break;
-          default:
-            throw new RuntimeException( "Argument of TRUNC of date has to be between 0 and 5" );
-        }
+    // 2 arguments: truncation of dates to a certain precision
+    //
+    if ( ArgList.length == 2 ) {
+      if ( isNull( ArgList[0] ) ) {
+        return null;
+      } else if ( isUndefined( ArgList[0] ) ) {
+        return undefinedValue;
+      }
 
-        return cal.getTime();
-      } else {
-        throw new RuntimeException( "The function call truncDate requires 2 arguments: a date and a level (int)" );
+      // This is the truncation of a date...
+      // The second argument specifies the level: ms, s, min, hour, day, month, year
+      //
+      java.util.Date dArg1 = null;
+      Integer level = null;
+      try {
+        dArg1 = (java.util.Date) ArgList[0];
+        level = (Integer) ArgList[1];
+      } catch ( Exception e ) {
+        throw new RuntimeException( e.toString() );
       }
-    } catch ( Exception e ) {
-      throw new RuntimeException( e.toString() );
+      return ScriptAddedFunctions.truncDate( dArg1, level );
+    } else {
+      throw new RuntimeException( "The function call truncDate requires 2 arguments: a date and a level (int)" );
     }
   }
 
+  @VisibleForTesting
+  static Date truncDate( Date dArg1, Integer level ) {
```

Comment:
This method appears to be exactly the same as the method in the other class. Any reason it can't be moved to some common utility class so the code doesn't have to be duplicated?

Reply:
No reason at all. Was just continuing the bifurcation of the two steps (deprecated, non-deprecated). Didn't figure it was valuable to refactor things out of their silos since one is deprecated. If preferred, I can consolidate the methods to one of the utility classes (goodness knows there are a few of them in PDI) and call the common one from both.

Reply:
Fair enough, and clearly not a huge deal.

Reply:
Fair enough, and clearly not a huge deal.